### PR TITLE
[mac] Add user-framework support for Xamarin.Mac

### PR DESCRIPTION
**Pull request by spouliot**
and add test project for xharness, which required moving some iOS stuff

### CODE REVIEW

File: tests/framework-test/iOS/FrameworkTests.cs [hunk 1]
```diff
@@ -28,8 +27,10 @@ public class FrameworkTests {
 		public void CFunction ()
 		{
 			Assert.AreEqual (42, CFunctions.theUltimateAnswer (), "a");
+#if !__MACOS__
```
Comment (rolfbjarne) [inline tests/framework-test/iOS/FrameworkTests.cs:30]: Why was it necessary to exclude these two functions?

**Review by rolfbjarne — APPROVED** [review]

File: tests/framework-test/iOS/FrameworkTests.cs (hunk 1 above)
Comment (spouliot) [reply tests/framework-test/iOS/FrameworkTests.cs:30]: They are inside the static, not dynamic, binaries and `codesign` does not sign them the same way, i.e. no `_CodeSignature` directory, the signature is embedded in the binary. 

While code signing _works_, verification (by us or the OS) fails.

**Comment by spouliot** [comment]
lack of space on bot :(

**Comment by spouliot** [comment]
build

**Review by dalexsoto — APPROVED** [review]

**Comment by rolfbjarne** [comment]
build

**Comment by spouliot** [comment]
lack of space on bot :(

**Comment by spouliot** [comment]
build

**Comment by spouliot** [comment]
build

**Review by tj-devel709 — APPROVED** [review]

**Review by emaf — APPROVED** [review]

**Pull request merged**